homeassistant/components/remootio/config_flow.py

```diff
@@ -0,0 +1,184 @@
+"""Config flow for Remootio integration."""
+from __future__ import annotations
+
+import logging
+import re
+from typing import Any
+
+from aioremootio import ConnectionOptions
+import voluptuous as vol
+from voluptuous.error import RequiredFieldInvalid
+from voluptuous.schema_builder import REMOVE_EXTRA
+
+from homeassistant import config_entries
+from homeassistant.components.cover import DEVICE_CLASS_GARAGE, DEVICE_CLASS_GATE
+from homeassistant.const import CONF_DEVICE_CLASS, CONF_IP_ADDRESS
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import (
+    CONF_API_AUTH_KEY,
+    CONF_API_SECRET_KEY,
+    CONF_SERIAL_NUMBER,
+    CONF_TITLE,
+    DOMAIN,
+)
+from .exceptions import CannotConnect, InvalidAuth, UnsupportedRemootioDeviceError
+from .utils import get_serial_number
+
+_LOGGER = logging.getLogger(__name__)
+
+REGEX_IP_ADDRESS = re.compile(
+    r"^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$"
+)
+REGEX_CREDENTIAL = re.compile(r"[0-9A-Z]{64}")
+
+INPUT_VALIDATION_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_IP_ADDRESS, msg="IP address is required"): vol.All(
```

Comment:
> Why can't a hostname be used?

Because the device tells me only the IP address. See also the [Remootio Websocket API documentation](https://github.com/remootio/remootio-api-documentation), e.g. the [newest version (v3)](https://github.com/remootio/remootio-api-documentation/blob/master/websocket_api_v3_specification.md) of it.

Reply:
A hostname for any IP address can be set on your DNS server/router. Advanced users might want to do that to keep control of their systems.

Reply:
> A hostname for any IP address can be set on your DNS server/router. Advanced users might want to do that to keep control of their systems.

I created this integration with the idea that even non-advanced users could use it.

The user has to reserve a fixed IP address for the device on his DHCP server or router so that he can integrate it. This can be a challenge for a non-advanced user.

Reply:
One does not preclude the other... Normally you can enter either a hostname or IP...

Reply:
I will adjust it.

Reply:
Fixed.